ProcessMaker/processmaker
# FOUR-10484 | Front End: Implement Asset Management UI

# Issue
Ticket: [FOUR-10484](https://processmaker.atlassian.net/browse/FOUR-10484)

- Develop the frontend components for Template Asset Management.
- Implement data retrieval and handling for asset management.
- Implement confirmation modals
	- Set up thye basic structure and layout of the modal.
	- Implement the frontend components and interactions.
	- Integrate the modal with the existing system and data.

# How to Test
1. Go to branch `task/FOUR-10484` in `processmaker`.
2. Go Designer → Processes. Select +PROCESS.
	- Select a template to create a Process with. You should select a template that you've already used to create a process in your environment. 
		- If you don't have a process in your environment that was created from a template, create a process from a template. Then repeat this step to create a process from the same template.
3. In the Create Process modal, select 'Save'. You should be taken to the Template Assets View at "/template/assets".
4. The assets associated with the template should be listed in their approriate tables.
5. Make changes to the assets, choosing to 'Update', 'Keep Previous' or 'Duplicate' them.
6. Select 'Continue' at the bottom of the page.
7. Select 'Yes' from the Confirmation modal. You should see a loading modal.
8. Once the Process has been created, you should see a 'Process Created Successfully' modal.
9. Select 'OK' and you should be taken to the modeler for the created process.
10. Go look at your assets to ensure that the changes you selected have been made.

ci:next
ci:deploy

# Code Review Checklist


[FOUR-10484]: https://processmaker.atlassian.net/browse/FOUR-10484?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
